pkg/sawmill/sawmill.go

```diff
@@ -0,0 +1,90 @@
+package sawmill
+
+// If you're wondering why this package is named sawmill, I do too: https://clips.twitch.tv/PlacidOutstandingPelicanCharlieBitMe
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"cloud.google.com/go/logging"
+	"google.golang.org/api/option"
+	logpb "google.golang.org/genproto/googleapis/logging/v2"
+)
+
+// Logger holds the newly generated error logging client, so that it can be used for our own, more simple custom logging functions instead of the ones the Google provides
+type Logger struct {
+	// The Logger
+	GoogleLogger *logging.Logger
+	// If their was an error with the initialization, it won't log to the cloud
+	InitError bool
+	// The service that logging was created for
+	ServiceName string
+}
+
+// InitClient creates a logging client
+func InitClient(projectID string, credentialsJSON string, environment string, serviceName string) (Logger, error) {
+	// The client takes the credentials in a byte array, so we do that conversion here
+	credentialsByte := []byte(credentialsJSON)
+	ctx := context.Background()
+
+	// Initializes an Google logging client
+	loggingClient, err := logging.NewClient(ctx, projectID, option.WithCredentialsJSON(credentialsByte)) // WithCredentialsJSON takes a raw JSON string in a byte array
+
+	if err != nil {
+		// If there is an error with initialization, it returns a Client object containing an empty ErrorClient, so nothing will get logged to the cloud, only the console
+		return Logger{nil, true, ""}, err
+	}
+
+	loggingClient.OnError = func(err error) {
+		log.Printf("Logging error in "+serviceName+": %v", err)
+	}
+
+	// Creates the actual logger
+	logger := loggingClient.Logger(serviceName)
+
+	// If we are in a dev environment, we don't want to log to the cloud, so this variable is used as the value for InitError, and if is set to true, nothing will be logged to the cloud
+	var isDev bool = false
+
+	// Here the aforementioned variable is set to true if we are in the DEV environment
+	if environment == "DEV" {
+		isDev = true
+	}
+
+	return Logger{logger, isDev, serviceName}, nil
```

Comment:
To clarify, at this point `isDev`  == `InitError`?

If this is the case, this seems wrong. We should add a property to logger called `isDevEnv` or something like that and set that property to true or false. I know we talked about using `InitError`, but I see down the line us running into issues when we want to do things based on these two scenarios

Reply:
Ok, I'll add that property.